roles/version_update_single_node/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+    SC_TIMEOUT: "{{ sc_timeout }}"
+
+  block:
+    - name: Get all available VMs
+      scale_computing.hypercore.vm_info:
+      register: vms
+    - ansible.builtin.debug:
+        var: vms
+
+    - name: Shutdown VMs
+      scale_computing.hypercore.vm:
+        vm_name: "{{ item.vm_name }}"
+        power_state: shutdown
+      loop: "{{ vms }}"
+      register: vm_shutdown_result
+
+    - name: Get available updates
+      scale_computing.hypercore.api:
+        action: get
+        endpoint: /rest/v1/Update
+      register: available_updates
+
+    - name: Update HC3
+      scale_computing.hypercore.api:
+        action: post
+        endpoint: "/rest/v1/Update/{{ item.uuid }}/apply"
+      when: available_updates
+      loop: "{{ available_updates }}"
```

Comment:
I am not really sure how it's those ``available_updates`` are sorted...
On our NUCs, there is only one update available, so it will show only ``9.1.23.210897``, on others, well, there are no updates available at all, because those NUCs are on the latest (not-yet-official) version ``9.2.x``, so the list is empty.

Reply:
Oh and if we have an update available, let's say ``9.1.23.210897``, which is also a uuid of an update, the update path/endpoint would be ``rest/v1/Update/9.1.23.210897/apply``.

Reply:
Yes, all true. We will decide how to filter available updates after we know could be there.